py/_io/terminalwriter.py

```diff
@@ -137,10 +137,17 @@ def __init__(self, file=None, stringio=False, encoding=None):
             file = colorama.AnsiToWin32(file).stream
         self.encoding = encoding or getattr(file, 'encoding', "utf-8")
         self._file = file
-        self.fullwidth = get_terminal_width()
         self.hasmarkup = should_do_markup(file)
         self._lastlen = 0
 
+    @property
+    def fullwidth(self):
+        return getattr(self, '_terminal_width', get_terminal_width())
```

Comment:
this one will in every case call get_terminal_width
please avoid that in case the attribute exists

the original proposal was to call if it is none and to have the attribute in any case